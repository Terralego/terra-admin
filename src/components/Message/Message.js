import React from 'react';
import { Callout } from '@blueprintjs/core';

const Message = ({ intent = 'warning', children, ...props }) => !!children && (
  <Callout intent={intent} {...props}>
    {children}
  </Callout>
);

export default Message;
