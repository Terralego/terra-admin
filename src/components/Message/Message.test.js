import React from 'react';
import renderer from 'react-test-renderer';

import Message from './Message';

it('should render correctly', () => {
  const tree = renderer.create((
    <Message>
      Lorem ipsum dolor sit amet.
    </Message>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should not render anything', () => {
  const tree = renderer.create((
    <Message />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
