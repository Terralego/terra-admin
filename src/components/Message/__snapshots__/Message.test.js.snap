// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render anything 1`] = `null`;

exports[`should render correctly 1`] = `
<div
  className="bp3-callout bp3-intent-warning bp3-callout-icon"
>
  <span
    className="bp3-icon bp3-icon-warning-sign"
    icon="warning-sign"
  >
    <svg
      data-icon="warning-sign"
      height={20}
      viewBox="0 0 20 20"
      width={20}
    >
      <desc>
        warning-sign
      </desc>
      <path
        d="M19.86 17.52l.01-.01-9-16-.01.01C10.69 1.21 10.37 1 10 1s-.69.21-.86.52l-.01-.01-9 16 .01.01c-.08.14-.14.3-.14.48 0 .55.45 1 1 1h18c.55 0 1-.45 1-1 0-.18-.06-.34-.14-.48zM11 17H9v-2h2v2zm0-3H9V6h2v8z"
        fillRule="evenodd"
      />
    </svg>
  </span>
  Lorem ipsum dolor sit amet.
</div>
`;
